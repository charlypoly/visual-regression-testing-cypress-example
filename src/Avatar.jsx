import * as React from "react";

const Avatar = () => {
  return <div>coucou</div>;
};

export default Avatar;
