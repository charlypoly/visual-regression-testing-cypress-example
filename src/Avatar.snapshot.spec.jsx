import React from "react";
import renderer from "react-test-renderer";
import Avatar from "../src/Avatar";

describe("<Avatar />", () => {
  describe("with a valid image", () => {
    it("renders correctly", () => {
      const tree = renderer
        .create(
          <Avatar
            src={
              "https://i.picsum.photos/id/623/200/200.jpg?hmac=xquTjHIYmAPV3XEGlIUaV_KWyEofkbortxrK79jJhWA"
            }
          />
        )
        .toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
  describe("with an invalid image", () => {
    it("renders correctly", async () => {
      const tree = renderer.create(<Avatar src={"<not_and_url>"} />).toJSON();
      expect(tree).toMatchSnapshot();
    });
  });
});
