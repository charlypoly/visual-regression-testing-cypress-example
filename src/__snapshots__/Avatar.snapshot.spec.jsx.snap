// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Avatar /> with a valid image renders correctly 1`] = `
<div
  className="sc-bdvvtL oSqWo"
>
  <img
    onError={[Function]}
    src="https://i.picsum.photos/id/623/200/200.jpg?hmac=xquTjHIYmAPV3XEGlIUaV_KWyEofkbortxrK79jJhWA"
  />
</div>
`;

exports[`<Avatar /> with an invalid image renders correctly 1`] = `
<div
  className="sc-bdvvtL oSqWo"
>
  <img
    onError={[Function]}
    src="<not_and_url>"
  />
</div>
`;
